elide-core/src/test/java/com/yahoo/elide/core/security/AggregationStorePermissionExecutorTest.java

```diff
@@ -0,0 +1,191 @@
+import com.yahoo.elide.core.type.ClassType;
+import com.yahoo.elide.core.type.Type;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import lombok.Value;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+
+/**
+ * test Aggregation Store Permission Executor
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class AggregationStorePermissionExecutorTest {
+    private EntityDictionary dictionary;
+    private ElideSettings elideSettings;
+
+
+    public static class FilterCheck extends FilterExpressionCheck<Object> {
+
+        @Override
+        public FilterExpression getFilterExpression(Type<?> entityClass, com.yahoo.elide.core.security.RequestScope requestScope) {
+            Path path = super.getFieldPath(entityClass, requestScope, "getFilterDim", "filterDim");
+            return new NotNullPredicate(path);
+        }
+    }
+
+    @BeforeAll
+    public void setup() {
+        Map<String, Class<? extends Check>> checks = new HashMap<>();
+        checks.put("user all", Role.ALL.class);
+        checks.put("user none", Role.NONE.class);
+        checks.put("filter check", FilterCheck.class);
+        dictionary = TestDictionary.getTestDictionary(checks);
+        elideSettings = new ElideSettingsBuilder(null)
+                .withEntityDictionary(dictionary)
+                .build();
+    }
+
+    @Test
+    public void testUserPermissions() {
+        @Entity
+        @Include
+        @Value
+        @ReadPermission(expression = "user none or filter check")
+        class Model {
+            String filterDim;
+            long metric;
+            long metric2;
+
+            @ReadPermission(expression = "user all")
+            public String getFilterDim() {
+                return filterDim;
+            }
+
+            @ReadPermission(expression = "user none")
+            public long getMetric() {
+                return metric;
+            }
+        }
+
+        com.yahoo.elide.core.RequestScope scope = bindAndgetRequestScope(Model.class);
+        PermissionExecutor executor = scope.getPermissionExecutor();
+
+        // evaluated expression = user all
+        Assertions.assertEquals(
+                ExpressionResult.PASS,
+                executor.checkUserPermissions(ClassType.of(Model.class), ReadPermission.class, "filterDim"));
+
+
+        // evaluated expression = user none -> ForbiddenAccess
+        Assertions.assertThrows(
+                ForbiddenAccessException.class,
+                () -> executor.checkUserPermissions(ClassType.of(Model.class), ReadPermission.class, "metric"));
+
+
+        // evaluated expression = (user none or filter check)
+        Assertions.assertEquals(
+                ExpressionResult.DEFERRED,
```

Comment:
This doesn't seem right to me.  The filter check should never apply at the field level.

Reply:
The field level check should not inherit the entity level check here.